realm/realm-library/src/main/java/io/realm/OrderedRealmCollection.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2016 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm;
+
+import java.util.List;
+
+/**
+ * A {@code OrderedRealmCollection} is a collection which maintains an ordering for its elements. Every
+ * element in the {@code OrderedRealmCollection} has an index. Each element can thus be accessed by its
+ * index, with the first index being zero. Normally, {@code OrderedRealmCollection}s allow duplicate
+ * elements, as compared to Sets, where elements have to be unique.
+ */
+public interface OrderedRealmCollection<E extends RealmObject> extends List<E>, RealmCollection<E> {
+
+    /**
+     * Gets the first object from the collection.
+     *
+     * @return the first object.
+     * @throws IndexOutOfBoundsException if the collection is empty.
+     */
+    E first();
+
+    /**
+     * Gets the last object from the collection.
+     *
+     * @return the last object.
+     * @throws IndexOutOfBoundsException if the collection is empty.
+     */
+    E last();
+
+    /**
+     * Sorts a collection based on the provided field in ascending order.
+     *
+     * @param fieldName the field name to sort by. Only fields of type boolean, short, int, long, float, double, Date,
+     *                  and String are supported.
+     * @return A sorted view of the collection.
+     * @throws java.lang.IllegalArgumentException if field name does not exist or it has an invalid type.
+     * @throws java.lang.IllegalStateException if the Realm is closed, called on the wrong thread or the collection is
+     *                                         an un-managed collection.
+     */
+    RealmResults<E> sort(String fieldName);
```

Comment:
I had the same thoughts, but the problem is that the underlying type _is_ RealmResults and RealmResults are quite restricted in terms of what you can do with it. So exposing RealmResults as the returned type has two benefits:

1) It is really clear that we return a view of the data and not some modified version of the original.
2) The methods not supported on the returned collection is well-defined.